File: client/src/types/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseSetModel.ts
import type { ExerciseSetType } from "../Enums/ExerciseSetType";

export interface WorkoutTemplateExerciseSetModel
{
	id: number;
	orderIndex: number;
	setType: ExerciseSetType;
	weightKg?: number;
	reps?: number;
	durationSeconds?: number;
	distanceMeters?: number;
	rpe?: number;
	restSeconds?: number;
	notes?: string;
}

File: server/FitMate.Core/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseSetModel.cs
namespace FitMate.Core.JsonModels.WorkoutTemplates;

public class WorkoutTemplateExerciseSetModel
{
    public long Id { get; set; }
    public int OrderIndex { get; set; }
    public decimal? WeightKg { get; set; }
    public int? Reps { get; set; }
    public int? DurationSeconds { get; set; }
    public decimal? DistanceMeters { get; set; }
    public decimal? Rpe { get; set; }
    public int? RestSeconds { get; set; }
    public string? Notes { get; set; }
}
